packages/contracts/contracts/L1/rollup/CanonicalTransactionChain.sol

```diff
@@ -365,18 +360,13 @@ contract CanonicalTransactionChain is ICanonicalTransactionChain, Lib_AddressRes
             "Not enough BatchContexts provided."
         );
 
-        // Take a reference to the queue and its length so we don't have to keep resolving it.
-        // Length isn't going to change during the course of execution, so it's fine to simply
-        // resolve this once at the start. Saves gas.
-        IChainStorageContainer queueRef = queue();
-        uint40 queueLength = _getQueueLength(queueRef);
-
         // Counter for number of sequencer transactions appended so far.
         uint32 numSequencerTransactions = 0;
 
-        // We will sequentially append leaves which are pointers to the queue.
-        // The initial queue index is what is currently in storage.
-        uint40 nextQueueIndex = getNextQueueIndex();
+        // Cache the nextQueueIndex.
+        // This is safe as long as nothing reads or writes to the nextQueueIndex
+        // storage variable until it is updated with the value from nextQueueIndexCached.
+        uint40 nextQueueIndexCached = nextQueueIndex;
```

Comment:
Open to suggestions for better variable names.

Reply:
`_nextQueueIndex` ?

`memoryNextQueueIndex`?

`inMemoryNextQueueIndex`?

`nextQueueIndexInMemory`?

I agree it would be nice to have a better named variable. And I'm right that this is "cached" to save gas?

Reply:
I ended up following [OZ' convention](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L41-L42) with a `_` prefix for private variables. 

So now the global instance is `_nextQueueIndex`, and the local ones are just `nextQueueIndex`. This has the added benefit of letting us also use `nextQueueIndex` as a local variable in `_appendBatch` and `_getBatchExtraData`.